zebra-state/src/arbitrary.rs

```diff
@@ -33,6 +33,32 @@ impl Prepare for Arc<Block> {
     }
 }
 
+impl<T> From<T> for ChainTipBlock
+where
+    T: Prepare,
+{
+    fn from(block: T) -> Self {
+        block.prepare().into()
+    }
+}
+
+impl From<PreparedBlock> for ChainTipBlock {
+    fn from(prepared: PreparedBlock) -> Self {
+        let PreparedBlock {
+            block: _,
+            hash,
+            height,
+            new_outputs: _,
+            transaction_hashes,
+        } = prepared;
+        Self {
+            hash,
+            height,
+            transaction_hashes,
+        }
```

Comment:
Question: I'm wondering if separating the binding from the construction is a matter of style or if there's some advantage I missed. Like, is this used to be able to capture when new fields are added to `PreparedBlock` in the future? Just for comparison, my default approach would be something like:

```suggestion
        Self {
            hash: prepared.hash,
            height: prepared.height,
            transaction_hashes: prepared.transaction_hashes,
        }
```

I think it's a matter of style, so there's no need to change it. I'm just curious if there's a reason or if there are certain situations where it's preferable to write the destructuring separately :thinking:

Reply:
It's used to avoid cloning the `Arc`s, which is slightly more expensive due to the atomic reference count.

I've also found it useful to explicitly show all the fields, and show which fields we're dropping.